brreg/src/main/kotlin/no/nav/helsearbeidsgiver/inntektsmelding/brreg/VirksomhetLoeser.kt

```diff
@@ -76,11 +79,33 @@ class VirksomhetLoeser(
             publishData(behov.createData(mapOf(DataFelt.VIRKSOMHET to navn)))
         } catch (ex: FantIkkeVirksomhetException) {
             logger.error("Fant ikke virksomhet for $orgnr")
-            publishFail(behov.createFail("Ugyldig virksomhet $orgnr"))
+            rapidsConnection.publish(
+                createFail(behov, "Fant ikke virksomhet")
+                    .also {
+                        logger.error("Publiserte feil for ${BehovType.HENT_IM_ORGNR}.")
+                        sikkerLogger.error("Publiserte feil:\n${it.parseJson().toPretty()}")
+                    }
+            )
         } catch (ex: Exception) {
             logger.error("Det oppstod en feil ved henting for $orgnr")
             sikkerLogger.error("Det oppstod en feil ved henting for orgnr $orgnr: ", ex)
-            publishFail(behov.createFail("Klarte ikke hente virksomhet"))
+            rapidsConnection.publish(
+                createFail(behov, "Klarte ikke hente virksomhet")
+                    .also {
+                        logger.error("Publiserte feil for ${BehovType.HENT_IM_ORGNR}.")
+                        sikkerLogger.error("Publiserte feil:\n${it.parseJson().toPretty()}")
+                    }
+            )
         }
     }
+
+    private fun createFail(behov: Behov, feilmelding: String): String {
```

Comment:
Hadde glømt bort dette litt. Men nå husker jeg det.
"Domene modellen"  med Behov ,Data , Event, Fail ble jo innført for å erstatte blandt annet Feil fra "felles"
(https://github.com/navikt/helsearbeidsgiver-inntektsmelding/pull/195)
Fail fra felles var noe som ble innført før "domenemodellen" fantes.
Stemmer for at vi fortsetter å bruke den her hvis det er mulig. For ellers blir det slik at vi bruker domenemodellen for Behov og Event, men ikke Fail. Gir det mening?

Reply:
Ja, det er mer konsistent. Endrer.